### ISSUE TITLE
concave_hull produces segmentation fault with specific combination of points

### ISSUE THREAD

**Issue by raine-fmg** [shapely/shapely#1873]
## Expected behavior and actual behavior.
concave_hull is causing a segmentation fault

## Steps to reproduce the problem.

```
from shapely import concave_hull, MultiPoint
import numpy as np

concave_hull(MultiPoint(np.array([[ 584245.72096874, 7549593.72686167],
       [584251.71398371, 7549594.01629478],
       [584242.72446125, 7549593.58214511],
       [584230.73978847, 7549592.9760418 ],
       [584233.73581213, 7549593.13045099],
       [584236.7318358 , 7549593.28486019],
       [584239.72795377, 7549593.43742855],
       [584227.74314188, 7549592.83423486]])))
```

## Operating system
Windows 10 Enterprise 22H2

## Shapely version and provenance
shapely==2.0.1 (installed via pip)
numpy==1.24.2 

## Aditional Info
I recreated in a fresh venv to ensure there were no other complexities associated with my environment. I originally posted the issue here on [SO](https://stackoverflow.com/questions/76919688/seg-fault-raised-by-shapely-concave-hull), just in case it was something geometric that I wasn't aware of. It seems to be something to do with collinearity (but I include more notes on removing certain points etc in my SO question).

**Comment by jorisvandenbossche** [comment]
This will be fixed in GEOS 3.12.1 (when released)

**Comment by mwtoews** [comment]
This is a duplicate of #1873 with upstream fixes on the way.

**Comment by mwtoews** [comment]
Also fixed in GEOS 3.11.4 (when released).

**Comment by mwtoews** [comment]
Fixed confirmed with shapely-2.0.5
